conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,256 @@ class CMakeFindPackageGenerator(Generator):
         endif()
         """)
 
+    find_components_tpl = Template(textwrap.dedent("""\
+        ########## MACROS ###########################################################################
+        #############################################################################################
+        {{ conan_message }}
+        {{ conan_find_apple_frameworks }}
+        {{ conan_package_library_targets }}
+
+        ########### FOUND PACKAGE ###################################################################
+        #############################################################################################
+
+        include(FindPackageHandleStandardArgs)
+
+        conan_message(STATUS "Conan: Using autogenerated Find{{ pkg_name }}.cmake")
+        set({{ pkg_name }}_FOUND 1)
+        set({{ pkg_name }}_VERSION "{{ pkg_version }}")
+        set({{ pkg_name }}_COMPONENTS {{ pkg_components }})
+
+        find_package_handle_standard_args({{ pkg_name }} REQUIRED_VARS
+                                          {{ pkg_name }}_VERSION VERSION_VAR {{ pkg_name }}_VERSION)
+        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
+
+        if({{ pkg_name }}_FIND_COMPONENTS)
+            foreach(_FIND_COMPONENT {{ '${'+pkg_name+'_FIND_COMPONENTS}' }})
+                list(FIND {{ pkg_name }}_COMPONENTS "{{ pkg_name }}::${_FIND_COMPONENT}" _index)
+                if(${_index} EQUAL -1)
+                    conan_message(FATAL_ERROR "Conan: Component '${_FIND_COMPONENT}' NOT found in package '{{ pkg_name }}'")
+                else()
+                    conan_message(STATUS "Conan: Component '${_FIND_COMPONENT}' found in package '{{ pkg_name }}'")
+                endif()
+            endforeach()
+        endif()
+
+        ########### VARIABLES #######################################################################
+        #############################################################################################
+
+        {%- for comp_name, comp in components %}
+
+        ########### COMPONENT {{ comp_name }} VARIABLES #############################################
+
+        set({{ pkg_name }}_{{ comp_name }}_INCLUDE_DIRS {{ comp.include_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_INCLUDE_DIR {{ comp.include_path }})
+        set({{ pkg_name }}_{{ comp_name }}_INCLUDES {{ comp.include_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_LIB_DIRS {{ comp.lib_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_RES_DIRS {{ comp.res_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_DEFINITIONS {{ comp.defines }})
+        set({{ pkg_name }}_{{ comp_name }}_COMPILE_DEFINITIONS {{ comp.compile_definitions }})
+        set({{ pkg_name }}_{{ comp_name }}_COMPILE_OPTIONS_LIST "{{ comp.cxxflags_list }}" "{{ comp.cflags_list }}")
```

Comment:
Is this a `;`-separated list?

Reply:
I've checked it works 👍 , I'd have preferred to use the same style, but it is ok

```suggestion
        set({{ pkg_name }}_{{ comp_name }}_COMPILE_OPTIONS_LIST "{{ comp.cxxflags_list }};{{ comp.cflags_list }}")
```

Reply:
I did not change the implementation of this variables. See https://github.com/conan-io/conan/blob/develop/conans/client/generators/cmake_find_package_common.py#L17